feat: add commit-hooks to expand Jira ticket# to URL

vharsh opened the pull request:
other significant changes:

- the commit-hook will enforce people to make changes against a Jira ticket
- commit messages must have a well defined type as defined by [conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0/)
- NOTE: this doesn't validate if the ticket exists, but helps adding one in the commit message for easier tracking if the same is provided


assumptions:

- the ticket belongs to the MOSIP sub-project in Jira(can be updated for different projects)
- the tickets for issues created in this repo would exist in MOSIP's Jira going forward & not on GitHub


Requirements/TODOs:

- This changeset would require a Jira ticket created, post which the commit-message would need an update
- Documenting the developer facing changes & necessary information to avoid surprises & prevent blockers
- Testing it out in IDE integration of `git-commit` at least from AndroidStudio, Vscode, Xcode


Tl; dr:

Sample commit messages for future

```bash
# earlier the commit-message would be reflected in git as-is, without any mutation
# post-this changeset the commit-message would have a Jira URL to improve code & ticket co-relation

$ git commit -m 'feat(#1234): improve visibility of button on low-res device'
$ # the above commit would look like this in real life
$ git log

commit <hash>
Author: John Doe <john.doe@example.com>
Date:   Tue Apr 18 11:57:54 2023 +0530

    feat(#1234): improve visibility of button on low-res device

    Link: https://mosip.atlassian.net/browse/MOSIP-1234
```





Co-authored-by: KiruthikaJeyashankar <81218987+KiruthikaJeyashankar@users.noreply.github.com>

vharsh committed:
feat: add commit-hooks to expand Jira ticket# to URL

other significant changes:

- the commit-hook will enforce people to make changes against a known Jira ticket
- commit messages must have a well defined type as defined by conventionalcommits

assumption: the ticket belongs to the MOSIP sub-project in Jira

Co-authored-by: Kiruthika Jeyashankar <kiruthika.j@thoughtworks.com>

vharsh commented:
A similar changeset will be raised against the release branch if this is approved & accepted.

KiruthikaJeyashankar commented on .husky/commit-msg:
```diff
@@ -0,0 +1,36 @@
+if [[ "$ELIGIBLE" -eq 1 ]]; then
+    ISSUETYPE_TICKET=$( echo $MESSAGE | cut -d ':' -f1 ) # `feat(#1234)`
+    ISSUENUMBER=$( echo $ISSUETYPE_TICKET | grep -Eo '[0-9]+') # `1234`
+    # TODO: Check if the ISSUENUMBER is a numeric entity as GitHub issues are numeric
```
Shall we remove this TODO, as this is already performed

KiruthikaJeyashankar committed:
fix: Display log if commit has no ticket number

Display log if the provided commit message's prefix is supposed to have ticket number

KiruthikaJeyashankar committed:
refactor: Remove unused comments
